buffer/src/main/java/io/netty/buffer/AdvancedLeakAwareByteBuf.java

```diff
@@ -926,25 +929,50 @@ public ByteBuf asReadOnly() {
     @Override
     public ByteBuf retain() {
         leak.record();
-        return super.retain();
+        try {
+            return super.retain();
+        } catch (IllegalReferenceCountException e) {
+            throw dealWithIllegalReferenceCountException(e);
+        }
     }
 
     @Override
     public ByteBuf retain(int increment) {
         leak.record();
-        return super.retain(increment);
+        try {
+            return super.retain(increment);
```

Comment:
Seems like other operation(like `getBytes`) also need to catch `IllegalReferenceCountException`.